solutions/gold/ac-distancequeries.mdx

```diff
@@ -0,0 +1,182 @@
+---
+id: ac-distancequeries
+source: AC Beginner Contest 294
+title: Distance Queries on a Tree
+author: Justin Ji
+---
+
+<Spoiler title = "Hint">
+
+Root the tree. For an edge from $u$ -> $v$, where $u$ is
+the parent, we place the edge weight at node $u$.
+
+</Spoiler>
+
+<Spoiler title = "Hint 2">
+
+Given the previous hint, we can efficiently calculate the distance from the root
+to any node in the tree. How can we use this to find the distance from any pair
+of nodes?
+
+</Spoiler>
+
+<Spoiler title = "Answer to Hint 2">
+
+The distance between any pair of nodes $(u, v)$ is
+$dist(u) + dist(v) - 2 \cdot dist(lca(u, v))$,
+where $dist(x)$ equals the distance from node $x$ to the
+root.
```

Comment:
i think a better word is depth?

Reply:
depth typically refers to the "level" of the node no? the graph is weighted so i think distance is more appropriate here

Reply:
yeah i uh, agree with justin here

Reply:
oops yeah you're right